packages/back-end/src/jobs/informationSchema.ts

```diff
@@ -0,0 +1,75 @@
+import Agenda, { Job } from "agenda";
+import { initializeDatasourceInformationSchema } from "../services/datasource";
+import { logger } from "../util/logger";
+import { getDataSourceById } from "../models/DataSourceModel";
+import {
+  getInformationSchemaByDatasourceId,
+  updateInformationSchemaById,
+} from "../models/InformationSchemaModel";
+
+const CREATE_INFORMATION_SCHEMA_JOB_NAME = "createInformationSchema";
+type CreateInformationSchemaJob = Job<{
+  datasourceId: string;
+  organization: string;
+}>;
+
+let agenda: Agenda;
+export default function (ag: Agenda) {
+  agenda = ag;
+
+  agenda.define(
+    CREATE_INFORMATION_SCHEMA_JOB_NAME,
+    async (job: CreateInformationSchemaJob) => {
+      const { datasourceId, organization } = job.attrs.data;
+
+      if (!datasourceId || !organization) return;
+
+      const datasource = await getDataSourceById(datasourceId, organization);
+
+      if (datasource?.type !== "postgres" && datasource?.type !== "bigquery")
```

Comment:
So, currently, without this check, we are trying to build the informationSchema, and when we find that the `integration.getInformationSchema` method isn't defined, we're throwing an error for "Datasource not supported". However, now that we have changed the `initializeDatasourceInformationSchema` to create a blank informationSchema whose `id` is then stored on the `datasource` object - whenever we do add support for these datasources down the road, when the user goes to build the informationSchema, they'll see an error that this datasource type isn't supported. 

Which, at the time they created the datasource was true, but is now no-longer true. Adding this check, prevents all of that, and instead, when we add support for a datasource later down the road, the user can then build an informationSchema without getting an old, out of date error message."

This isn't them most elegant check, and will get cumbersome as time goes on, but it will prevent a bad UX. 

If we don't want this check, we'll just need to re-think the front-end experience for this use case.

I just want to prevent us from having to run some sort of database migration to clear errors for now supported datasource types.

Reply:
The problem stems from the fact that all errors are treated equally when they probably shouldn't be.  Instead of just storing a generic error string, if we also stored the type of error, then we would be able to avoid checks like this in the code and still provide a really good user experience.

One of the simplest ways to do this is to make separate Error classes for different types of errors.

```ts
class DataSourceNotSupportedError extends Error {
  constructor() {
    super("This data source is not supported yet");
    this.name = "DataSourceNotSupportedError";
  }
}

class DataSourceTooBigError extends Error {
  constructor() {
    super("We don't support data sources this large yet");
    this.name = "DataSourceTooBigError";
  }
}
```

Then, in your code you would `throw new DataSourceNotSupportedError()` for example.

In the code that catches the error, you can figure out which type was thrown:

```ts
catch(e) {
  if (e instanceof DataSourceNotSupportedError) {
    
  } else ...
}
```